identity/src/main/java/com/stripe/android/identity/navigation/ConfirmationFragment.kt

```diff
@@ -31,58 +32,45 @@ internal class ConfirmationFragment(
         identityViewModelFactory
     }
 
-    private lateinit var binding: ConfirmationFragmentBinding
-
     override fun onCreateView(
         inflater: LayoutInflater,
         container: ViewGroup?,
         savedInstanceState: Bundle?
-    ): View {
-        binding = ConfirmationFragmentBinding.inflate(inflater, container, false)
-        binding.kontinue.setOnClickListener {
-            lifecycleScope.launch {
-                identityViewModel.analyticsState.collectLatest {
-                    identityViewModel.sendAnalyticsRequest(
-                        identityViewModel.identityAnalyticsRequestFactory.verificationSucceeded(
-                            isFromFallbackUrl = false,
-                            scanType = it.scanType,
-                            requireSelfie = it.requireSelfie,
-                            docFrontRetryTimes = it.docFrontRetryTimes,
-                            docBackRetryTimes = it.docBackRetryTimes,
-                            selfieRetryTimes = it.selfieRetryTimes,
-                            docFrontUploadType = it.docFrontUploadType,
-                            docBackUploadType = it.docBackUploadType,
-                            docFrontModelScore = it.docFrontModelScore,
-                            docBackModelScore = it.docBackModelScore,
-                            selfieModelScore = it.selfieModelScore
+    ) = ComposeView(requireContext()).apply {
+        setViewCompositionStrategy(ViewCompositionStrategy.DisposeOnViewTreeLifecycleDestroyed)
+        setContent {
+            val verificationPage by identityViewModel.verificationPage.observeAsState()
```

Comment:
Now that we use this in Compose, maybe it makes sense to expose `verificationPage` as a `StateFlow<Resource<VerificationPage>>`? This way, we can give it `Resource.loading()` as an initial value and don’t have to deal with nullability.

Reply:
Sure, added this in [this](https://jira.corp.stripe.com/browse/IDPROD-4581) internal ticket. The `verificationPage` is currently used in multiple fragments, will make that change when we get rid off all fragments